WcaOnRails/app/models/round.rb

```diff
@@ -121,6 +121,10 @@ def advancement_condition_to_s(short: false)
     advancement_condition ? advancement_condition.to_s(self, short: short) : ""
   end
 
+  def advancement_condition_is_valid?
+    final_round? || advancement_condition?.present?
```

Comment:
Here you can go `advancement_condition.present?`, or simply `advancement_condition`, because it evaluates to "something" which is not false.